lib/service/InitService.ts

```diff
@@ -0,0 +1,74 @@
+import errors from '../service/errors';
+import { EventEmitter } from 'events';
+import SwapClientManager from '../swaps/SwapClientManager';
+import NodeKey from '../nodekey/NodeKey';
+import assert = require('assert');
+
+interface InitService {
+  once(event: 'nodekey', listener: (nodeKey: NodeKey) => void): this;
+  emit(event: 'nodekey', nodeKey: NodeKey): boolean;
+}
+
+class InitService extends EventEmitter {
+  /** Whether there is a pending `CreateNode` or `UnlockNode` call. */
+  public pendingCall = false;
+
+  constructor(private swapClientManager: SwapClientManager, private nodeKeyPath: string, private nodeKeyExists: boolean) {
+    super();
+  }
+
+  public createNode = async (args: { password: string }) => {
+    const { password } = args;
+    if (this.nodeKeyExists) {
+      throw errors.UNIMPLEMENTED;
+    }
+    if (this.pendingCall) {
+      throw errors.PENDING_CALL_CONFLICT;
+    }
+
+    this.pendingCall = true;
+    const seed = await this.swapClientManager.genSeed();
+    let nodeKey: NodeKey;
+
+    if (seed) {
+      const seedBytes = typeof seed.encipheredSeed === 'string' ?
```

Comment:
I actually mainly wrote it this way for the compiler. According to the type definitions we're using currently, `encipheredSeed` (or any field that's a `bytes` protobuf type) will have a `string | Uint8Array` type. I'm not sure why this is - it always seems to be a base64 string in practice. However, I figured it made sense to check the type just in case.